contracts/withdrawManager.sol

```diff
@@ -47,67 +47,82 @@ contract WithdrawManager {
         withdrawTxClaimed = new bool[][](governance.MAX_TXS_PER_BATCH());
     }
 
-    /**
-     * @notice Allows user to withdraw the balance in the leaf of the balances tree.
-     *        User has to do the following: Prove that a transfer of X tokens was made to the burn address or leaf 0
-     *        The batch we are allowing withdraws from should have been already finalised, so we can assume all data in the batch to be correct
-     * @param _batch_id Deposit tree depth or depth of subtree that is being deposited
-     * @param withdraw_tx_proof contains the siblints, txPath and the txData for the withdraw transaction
-     */
-    function Withdraw(
-        uint256 _batch_id,
-        Types.CommitmentInclusionProof memory commitmentMP,
-        Types.PDAMerkleProof memory _pda_proof,
-        Types.TransactionMerkleProof memory withdraw_tx_proof
-    ) public {
-        Types.Batch memory batch = rollup.getBatch(_batch_id);
-
-        // check if the batch is finalised
-        require(block.number > batch.finalisesOn, "Batch not finalised yt");
-        // verify transaction exists in the batch
-        merkleUtils.verify(
-            commitmentMP.commitment.txRootCommitment,
-            RollupUtils.BytesFromTx(withdraw_tx_proof._tx.data),
-            withdraw_tx_proof._tx.pathToTx,
-            withdraw_tx_proof.siblings
-        );
-
-        // check if the transaction is withdraw transaction
-        // ensure the `to` leaf was the 0th leaf
-        require(
-            withdraw_tx_proof._tx.data.toIndex == 0,
-            "Not a withdraw transaction"
-        );
-
-        bool isClaimed = withdrawTxClaimed[_batch_id][withdraw_tx_proof
-            ._tx
-            .pathToTx];
-        require(!isClaimed, "Withdraw transaction already claimed");
-        withdrawTxClaimed[_batch_id][withdraw_tx_proof._tx.pathToTx] = true;
-
-        // withdraw checks out, transfer to the account in account tree
-        address tokenContractAddress = tokenRegistry.registeredTokens(
-            withdraw_tx_proof._tx.data.tokenType
-        );
-
-        // convert pubkey path to ID
-        uint256 computedID = merkleUtils.pathToIndex(
-            _pda_proof._pda.pathToPubkey,
-            governance.MAX_DEPTH()
-        );
-
-        require(
-            computedID == withdraw_tx_proof._tx.data.fromIndex,
-            "Pubkey not related to the from account in the transaction"
-        );
-
-        address receiver = RollupUtils.calculateAddress(
-            _pda_proof._pda.pubkey_leaf.pubkey
-        );
-
-        uint256 amount = withdraw_tx_proof._tx.data.amount;
-
-        IERC20 tokenContract = IERC20(tokenContractAddress);
-        require(tokenContract.transfer(receiver, amount), "Unable to trasnfer");
-    }
+    // /**
```

Comment:
Can we just remove the whole code block instead of commenting off the code?

Reply:
Do not want to forget that this needs to be fixed, the only thing that needs fixing in this is how to validate with TxHashCommitment while cheery picking a transaction.

Reply:
But lets open and issue and delete the codeblock?

Reply:
I was hoping we could fix this today itself.

Reply:
Yes, either fixing it today or creating an issue are all better than commenting off them.

Reply:
https://github.com/thehubbleproject/RedditHubble/issues/156